pkg/workloadmeta/collectors/internal/containerd/image_sbom_trivy.go

```diff
@@ -137,3 +138,57 @@ func (c *collector) extractSBOMWithTrivy(ctx context.Context, storedImage *workl
 
 	return nil
 }
+
+// updateSBOMMetadata updates entered SBOM with new metadata properties if the initial SBOM status was successful
+// and there are new repoTags and repoDigests missing in the SBOM. It returns the updated SBOM.
+func updateSBOMMetadata(sbom *workloadmeta.SBOM, repoTags, repoDigests []string) *workloadmeta.SBOM {
+	if sbom.Status != workloadmeta.Success || sbom.CycloneDXBOM.Metadata.Component.Properties == nil {
+		return sbom
+	}
+
+	properties := *sbom.CycloneDXBOM.Metadata.Component.Properties
+	propertySet := buildPropertySet(properties)
+
+	properties = appendMissingProperties(properties, propertySet, repoTags, trivydx.PropertyRepoTag)
+	properties = appendMissingProperties(properties, propertySet, repoDigests, trivydx.PropertyRepoDigest)
+
+	sbom.CycloneDXBOM.Metadata.Component.Properties = &properties
+
+	return sbom
+}
+
+// buildPropertySet generates a set of properties.
+func buildPropertySet(properties []cyclonedx.Property) map[cyclonedx.Property]struct{} {
+	propertySet := make(map[cyclonedx.Property]struct{})
+	for _, property := range properties {
+		propertySet[property] = struct{}{}
+	}
+	return propertySet
+}
+
+// containsProperty function checks if a specified property is present in the property set.
+func containsProperty(propertySet map[cyclonedx.Property]struct{}, property cyclonedx.Property) bool {
```

Comment:
I'm not sure if it's worth it to define a function just to check if a value exists on a map.

Reply:
Fixed !